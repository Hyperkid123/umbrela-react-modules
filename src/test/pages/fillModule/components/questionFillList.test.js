import React from 'react'
import QuestionFillList from '../../../../pages/fillModule/components/questionFillList';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Question fill list component', () => {
  const initialProps = {}
  beforeEach(() => {
    initialProps.questions = []
    initialProps.questionType = 'CloseQuestion'
    initialProps.filters = {};
  });

  it('should render empty', () => {
    const tree = shallow(
      <QuestionFillList {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render empty with image preview', () => {
    const tree = shallow(
      <QuestionFillList
        {...initialProps}
        questions={[{
          questionId: 1,
          options: [],
          url: 'foo.url',
          questionType: 'MediaQuestion'
        }]}
        filters={{}}
      />
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should return if question is filtered', () => {
    const wrapper = shallow(
      <QuestionFillList
        {...initialProps}
        questions={[{
          questionId: 1,
          options: []
        }]}
        filters={{}}
      />
    )
    expect(wrapper.instance().checkFiltered(1)).toBeFalsy()
    wrapper.setProps({filters: {"1": []}})
    expect(wrapper.instance().checkFiltered(1)).toBeFalsy()
    wrapper.setProps({filters: {"1": [1]}});
    expect(wrapper.instance().checkFiltered(1)).toBeTruthy()
  })

  it('should update state on handle open request', () => {
    const wrapper = shallow(
      <QuestionFillList {...initialProps}/>
    )
    wrapper.instance().handleOpenDialog('foo.com');
    expect(wrapper.state().openDialog).toBeTruthy();
    expect(wrapper.state().dialogImage).toEqual('foo.com')
  })

  it('should update state on handle close request', () => {
    const wrapper = shallow(
      <QuestionFillList {...initialProps}/>
    )
    wrapper.instance().handleCloseDialog();
    expect(wrapper.state().openDialog).toBeFalsy();
  })

  it('should render with customHelp', () => {
    const tree = shallow(
      <QuestionFillList {...initialProps} questions={[{
        customHelp: 'custom help',
        questionId: 1,
        options: []
      }]}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })
});
