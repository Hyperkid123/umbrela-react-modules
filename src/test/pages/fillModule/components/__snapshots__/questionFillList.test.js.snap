// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question fill list component should render empty 1`] = `
<WithStyles(GridWrapper)
  alignItems="center"
  container={true}
  justify="center"
  spacing={0}
>
  <ScrollToTopOnMount />
  <WithStyles(GridWrapper)
    container={true}
    item={true}
    spacing={0}
    style={
      Object {
        "marginTop": 10,
      }
    }
  />
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="image-dialog-title"
    >
      Náhled
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</WithStyles(GridWrapper)>
`;

exports[`Question fill list component should render empty with image preview 1`] = `
<WithStyles(GridWrapper)
  alignItems="center"
  container={true}
  justify="center"
  spacing={0}
>
  <ScrollToTopOnMount />
  <WithStyles(GridWrapper)
    container={true}
    item={true}
    spacing={0}
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <WithStyles(GridWrapper)
      id="question_1"
      item={true}
      key="1"
      xs={12}
    >
      <WithStyles(GridWrapper)
        alignItems="center"
        container={true}
        spacing={0}
      >
        <WithStyles(GridWrapper)
          item={true}
          style={
            Object {
              "display": "flex",
              "marginLeft": "auto",
              "marginRight": "auto",
              "maxWidth": "100%",
            }
          }
        >
          <styled.div
            matrix={false}
            questionList={true}
          >
            <styled.div
              questionList={true}
            >
              <styled.h3>
                <span>
                  1
                </span>
                <span />
              </styled.h3>
            </styled.div>
            <styled.div
              questionList={true}
            >
              <LazyLoad
                offset={0}
                once={false}
                overflow={false}
                resize={false}
                scroll={true}
                unmountIfInvisible={false}
              >
                <styled.img
                  alt="foo.url"
                  onClick={[Function]}
                  src="foo.url"
                />
              </LazyLoad>
              <styled.div />
            </styled.div>
          </styled.div>
        </WithStyles(GridWrapper)>
      </WithStyles(GridWrapper)>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="image-dialog-title"
    >
      Náhled
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</WithStyles(GridWrapper)>
`;

exports[`Question fill list component should render with customHelp 1`] = `
<WithStyles(GridWrapper)
  alignItems="center"
  container={true}
  justify="center"
  spacing={0}
>
  <ScrollToTopOnMount />
  <WithStyles(GridWrapper)
    container={true}
    item={true}
    spacing={0}
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <WithStyles(GridWrapper)
      id="question_1"
      item={true}
      key="1"
      xs={12}
    >
      <WithStyles(GridWrapper)
        alignItems="center"
        container={true}
        spacing={0}
      >
        <WithStyles(GridWrapper)
          item={true}
          style={
            Object {
              "display": "flex",
              "marginLeft": "auto",
              "marginRight": "auto",
              "maxWidth": "100%",
            }
          }
        >
          <styled.div
            matrix={false}
            questionList={true}
          >
            <styled.div
              questionList={true}
            >
              <styled.h3>
                <span>
                  1
                </span>
                <span />
              </styled.h3>
            </styled.div>
            <styled.div
              questionList={true}
            >
              <styled.div>
                <span
                  key="0"
                >
                  custom help
                  <br />
                </span>
              </styled.div>
            </styled.div>
          </styled.div>
        </WithStyles(GridWrapper)>
      </WithStyles(GridWrapper)>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogTitle)
      id="image-dialog-title"
    >
      Náhled
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</WithStyles(GridWrapper)>
`;
